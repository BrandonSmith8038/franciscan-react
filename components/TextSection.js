import React from 'react'
import { withStyles } from 'material-ui/styles'
import Typography from 'material-ui/Typography'
import InvertedButton from './InvertedButton'
import Grid from 'material-ui/Grid'

const styles = theme => ({
  section: {
    display: 'flex',
    alignItems: 'start',
    justifyContent: 'center',
    flexDirection: 'column',
    padding: '32px 0',
    backgroundColor: '#fff'
  },
  innerContainer: {
    width: '70%',
    margin: '0 auto'
  },
  sectionText: {
    fontSize: '18px',
    fontWeight: 400,
    lineHeight: 1.5
  },
  heroBtn: {
    display: 'flex',
    justifyContent: 'start',
    marginTop: '16px'
  },
  invertedBtn: {
    color: '#21412a',
    backgroundColor: 'transparent',
    border: '2px #21412a solid',
    boxShadow: 'none'
  },
  white: {
    color: '#fff'
  }
})

class TextSection extends React.Component {
  render () {
    const {
      classes,
      sectionTitle,
      text,
      btnLink,
      btnText,
      textColor,
      bgColor
    } = this.props

    const Button = (
      <Grid item className={classes.heroBtn} xs={12}>
        <InvertedButton
          href={btnLink}
          title={btnText}
          className={classes.invertedBtn}
          color="secondary"
          style={{ width: '192px' }}
        >
          {btnText}
        </InvertedButton>
      </Grid>
    )

    return (
      <div className={classes.section}>
        <div className={classes.innerContainer}>
          <Typography type="display1" gutterBottom>
            {sectionTitle}
          </Typography>

          <Typography className={classes.sectionText} type="body1" gutterBottom>
            {text}
          </Typography>
          {btnLink && btnText ? Button : null}
          {textColor}
          {bgColor}
        </div>
      </div>
    )
  }
}

export default withStyles(styles)(TextSection)
