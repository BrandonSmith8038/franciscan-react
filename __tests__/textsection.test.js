/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import TextSection from '../components/TextSection'

describe('TextSection', () => {
  it('should be defined', () => {
    expect(TextSection).toBeDefined()
  })
  it('should render', () => {
    const wrapper = shallow(<TextSection />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
  it('should render text when passed in', () => {
    const wrapper = shallow(<TextSection text="Test text" />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
