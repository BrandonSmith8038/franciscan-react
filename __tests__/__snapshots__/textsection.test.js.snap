// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSection should render 1`] = `
<TextSection
  classes={
    Object {
      "heroBtn": "TextSection-heroBtn-4",
      "innerContainer": "TextSection-innerContainer-2",
      "invertedBtn": "TextSection-invertedBtn-5",
      "section": "TextSection-section-1",
      "sectionText": "TextSection-sectionText-3",
      "white": "TextSection-white-6",
    }
  }
/>
`;

exports[`TextSection should render text when passed in 1`] = `
<TextSection
  classes={
    Object {
      "heroBtn": "TextSection-heroBtn-4",
      "innerContainer": "TextSection-innerContainer-2",
      "invertedBtn": "TextSection-invertedBtn-5",
      "section": "TextSection-section-1",
      "sectionText": "TextSection-sectionText-3",
      "white": "TextSection-white-6",
    }
  }
  text="Test text"
/>
`;
